### PR TITLE
Adopt more smart pointers in WebCore/page

### PR DESCRIPTION
**Pull request by cdumez**
#### c66e393c17cab05d2b5b7b0f4fc928b1808b144d
<pre>
Adopt more smart pointers in WebCore/page
<a href="https://bugs.webkit.org/show_bug.cgi?id=269048">https://bugs.webkit.org/show_bug.cgi?id=269048</a>

Reviewed by Darin Adler.

* Source/WebCore/html/shadow/SpinButtonElement.h:
* Source/WebCore/page/AutoscrollController.cpp:
(WebCore::AutoscrollController::stopAutoscrollTimer):
(WebCore::AutoscrollController::updateAutoscrollRenderer):
(WebCore::AutoscrollController::updateDragAndDrop):
(WebCore::AutoscrollController::startPanScrolling):
(WebCore::AutoscrollController::autoscrollTimerFired):
* Source/WebCore/page/BarProp.cpp:
(WebCore::BarProp::visible const):
* Source/WebCore/page/CaptionUserPreferences.cpp:
(WebCore::CaptionUserPreferences::currentPage const):
(WebCore::CaptionUserPreferences::setUserPrefersCaptions):
(WebCore::CaptionUserPreferences::userPrefersSubtitles const):
(WebCore::CaptionUserPreferences::setUserPrefersSubtitles):
(WebCore::CaptionUserPreferences::userPrefersTextDescriptions const):
(WebCore::CaptionUserPreferences::setUserPrefersTextDescriptions):
(WebCore::CaptionUserPreferences::captionPreferencesChanged):
(WebCore::CaptionUserPreferences::updateCaptionStyleSheetOverride):
(WebCore::CaptionUserPreferences::pageGroup const):
* Source/WebCore/page/CaptionUserPreferences.h:
(WebCore::CaptionUserPreferences::pageGroup const): Deleted.
* Source/WebCore/page/CaptionUserPreferencesMediaAF.cpp:
(WebCore::CaptionUserPreferencesMediaAF::~CaptionUserPreferencesMediaAF):
(WebCore::CaptionUserPreferencesMediaAF::userPrefersCaptions const):
(WebCore::CaptionUserPreferencesMediaAF::userPrefersSubtitles const):
(WebCore::CaptionUserPreferencesMediaAF::userPrefersTextDescriptions const):
(WebCore::CaptionUserPreferencesMediaAF::setInterestedInCaptionPreferenceChanges):
(WebCore::CaptionUserPreferencesMediaAF::captionsWindowCSS const):
(WebCore::CaptionUserPreferencesMediaAF::captionsBackgroundCSS const):
(WebCore::CaptionUserPreferencesMediaAF::captionsTextColor const):
(WebCore::CaptionUserPreferencesMediaAF::captionStrokeWidthForFont const):
(WebCore::CaptionUserPreferencesMediaAF::captionsDefaultFontCSS const):
(WebCore::CaptionUserPreferencesMediaAF::platformPreferredLanguages):
(WebCore::CaptionUserPreferencesMediaAF::preferredAudioCharacteristics const):
(WebCore::trackDisplayName):
* Source/WebCore/page/CaptionUserPreferencesMediaAF.h:
* Source/WebCore/page/Chrome.cpp:
(WebCore::Chrome::protectedPage const):
(WebCore::Chrome::scroll):
(WebCore::Chrome::createWindow):
(WebCore::Chrome::runModal):
(WebCore::Chrome::runJavaScriptConfirm):
(WebCore::Chrome::runJavaScriptPrompt):
(WebCore::Chrome::mouseDidMoveOverElement):
(WebCore::Chrome::getToolTip):
(WebCore::Chrome::print):
(WebCore::Chrome::displayID const):
(WebCore::Chrome::windowScreenDidChange):
(WebCore::Chrome::registerPopupOpeningObserver):
(WebCore::Chrome::unregisterPopupOpeningObserver):
(WebCore::Chrome::notifyPopupOpeningObservers const):
* Source/WebCore/page/Chrome.h:
* Source/WebCore/page/ContextMenuContext.cpp:
(WebCore::ContextMenuContext::ContextMenuContext):
* Source/WebCore/page/ContextMenuContext.h:
* Source/WebCore/page/ContextMenuController.cpp:
(WebCore::ContextMenuController::page):
(WebCore::ContextMenuController::clearContextMenu):
(WebCore::prepareContextForQRCode):
(WebCore::ContextMenuController::maybeCreateContextMenu):
(WebCore::ContextMenuController::showContextMenu):
(WebCore::ContextMenuController::didDismissContextMenu):
(WebCore::openNewWindow):
(WebCore::insertUnicodeCharacter):
(WebCore::ContextMenuController::contextMenuItemSelected):
(WebCore::ContextMenuController::populate):
(WebCore::ContextMenuController::addDebuggingItems):
(WebCore::ContextMenuController::checkOrEnableIfNeeded const):
(WebCore::ContextMenuController::showContextMenuAt):
(WebCore::ContextMenuController::performPDFJSAction):
* Source/WebCore/page/ContextMenuController.h:
(WebCore::ContextMenuController::page): Deleted.
* Source/WebCore/page/PopupOpeningObserver.h:

Canonical link: <a href="https://commits.webkit.org/274374@main">https://commits.webkit.org/274374@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/45909cf3c63fdad1d9c00bd3b3ea9c9e7a645bfd

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/38921 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/17852 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/41269 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/41453 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/34638 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/41227 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/20716 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/15201 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/32610 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/39494 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/15022 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/33757 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/13055 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/13009 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/34678 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/42730 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/35321 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/35008 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/38846 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/13732 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/11330 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/37072 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/15337 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/14998 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5074 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/14825 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | |

### CODE REVIEW

File: Source/WebCore/page/CaptionUserPreferencesMediaAF.cpp [hunk 1]
```diff
@@ -721,8 +721,8 @@ static String trackDisplayName(const TrackBase& track)
 
     auto preferredLanguages = userPreferredLanguages(ShouldMinimizeLanguages::No);
     auto defaultLanguage = !preferredLanguages.isEmpty() ? preferredLanguages[0] : emptyString(); // This matches `defaultLanguage`.
-    auto currentLocale = adoptCF(CFLocaleCreate(kCFAllocatorDefault, defaultLanguage.createCFString().get()));
-    auto localeIdentifier = adoptCF(CFLocaleCreateCanonicalLocaleIdentifierFromString(kCFAllocatorDefault, trackLanguageIdentifier.createCFString().get()));
+    RetainPtr currentLocale = adoptCF(CFLocaleCreate(kCFAllocatorDefault, defaultLanguage.createCFString().get()));
```
Comment (darinadler) [inline Source/WebCore/page/CaptionUserPreferencesMediaAF.cpp:724]: I think implicit in this change, and the many like it, is a style guideline: Write `RetainPtr` explicitly to make the object lifetime more obvious than if you wrote `auto`. Presumably similar for other types. The main reason I haven’t been a fan of that rule in the past is that it doesn’t work when you don’t have a named local variable, and I often like the brevity of doing `adoptCF` in an expression in a function call without having to name the result, and other similar idioms. I figured that if we had a good quality checking system it would be just as easy to get it right without having to see the name of the type. But I am OK with us going this way, especially if it’s helping us do this better in practice.

File: Source/WebCore/page/ContextMenuController.cpp [hunk 2]
```diff
@@ -1795,8 +1799,7 @@ void ContextMenuController::showImageControlsMenu(Event& event)
 
 void ContextMenuController::performPDFJSAction(LocalFrame& frame, const String& action)
 {
-    auto* pdfDocument = dynamicDowncast<PDFDocument>(frame.ownerElement()->document());
-    if (pdfDocument)
+    if (RefPtr pdfDocument = dynamicDowncast<PDFDocument>(frame.ownerElement()->document()))
```
Comment (darinadler) [inline Source/WebCore/page/ContextMenuController.cpp]: I would have changed this to just `document` as a drive-by. My opinion is that the variable name doesn’t have to say what the type is, unless there’s some important distinction, like multiple variables pointing to the same object. It doesn’t make the code easier to read; words are easier to read than acronyms smashed together with words.

**Review by darinadler — APPROVED** [review]
Bummer that we keep needing to add includes; this work probably is leading to slower builds a file at a time.

File: Source/WebCore/page/ContextMenuController.cpp (hunk 2 above)
Comment (cdumez) [reply Source/WebCore/page/ContextMenuController.cpp]: Agreed. Fixing.
